src/Workflow/StepAggregator.php

```diff
@@ -80,7 +80,9 @@ public function __construct(Reader $reader, $name = null)
      */
     public function addStep(Step $step, $priority = null)
     {
-        $priority = null === $priority && $step instanceof PriorityStep ? $step->getPriority() : null;
+        if (null === $priority) {
+            $priority =  $step instanceof PriorityStep ? $step->getPriority() : null;
+        }
```

Comment:
@mattrq - might be a little nicer to read? Obviously, though, both with work as expected :-)